tools/flowc/manipulation/tail_call.flow

```diff
@@ -151,142 +142,205 @@ fcReturnExps(e : FiExp, acc : [FiExp]) -> [FiExp] {
     }
 }
 
+fcVarExistsInNonReturnExps(ex : FiExp, v : string, non_return: bool) -> bool {
```

Comment:
it's a bit funny to write a non-tail recursive function to detect the tail one :)